File: extractor-go/internal/domain/Map.go
package domain

type MapCoord struct {
	MapId string
	X     int
	Y     int
}

type Map struct {
	PreviousHistoryID NullableInt32
	HistoryID         NullableInt32
	Id                string
	FileVersion       int32
	Name              NullableString
	SpecialCode       NullableInt32
	Mp3Name           NullableString
	Npcs              []MapNpc
	Warps             []MapWarp
	Spawns            []MapSpawn
}

type NpcType string

const (
	NpcTypeDialog NpcType = "dialog"
	NpcTypeShop   NpcType = "shop"
)

type MapNpc struct {
	Type     NpcType
	SpriteId int
	Name1    LocalizableString
	Name2    string
	Location MapCoord
}

type SpawnType string

const (
	SpawnTypeNormal SpawnType = "normal"
	SpawnTypeBoss   SpawnType = "boss"
)

type MapSpawn struct {
	Type     SpawnType
	SpriteId int
	Name1    LocalizableString
	Name2    string
	Level    int
	Amount   int
	Element  int
	Size     int
	Race     int
}

type MapWarp struct {
	From     MapCoord
	To       MapCoord
	WarpType WarpType
	SpriteId int
	Name1    LocalizableString
	Name2    string
}

type WarpType string

const (
	WarpTypeCommon       WarpType = "common"
	WarpTypeFreeNpc      WarpType = "free_npc"
	WarpTypeKafraDts     WarpType = "kafra_dts"
	WarpTypeCoolEventDts WarpType = "cool_event_dts"
	WarpTypePaidNpc      WarpType = "paid_npc"
	WarpTypeAirport      WarpType = "airport"
)

File: ui/src/models/Map.ts
import type { LocalizableString } from "./LocalizableString";

export type Map = {
	HistoryID: string;

	FileVersion: number;

	Id: string;

	Name: string;

	SpecialCode: number;

	Mp3Name: string;

	Npcs: MapNpc[];

	Warps: MapWarp[];

	Spawns: MapSpawn[];
}

export type MapNpc = {
	Type: string;

	SpriteId: number;

	Name1: LocalizableString;

	Name2: string;

	Location: MapCoord;
}

export type MapCoord = {
	MapId: string;

	X: number;

	Y: number;
}

export type MapWarp = {
	From: MapCoord;

	To: MapCoord;

	WarpType: string;

	SpriteId: number;

	Name1: LocalizableString;

	Name2: string;
}

export type MapSpawn = {
	Type: string;

	SpriteId: number;

	Name1: LocalizableString;

	Name2: string;

	Level: number;

	Amount: number;

	Element: number;

	Size: number;

	Race: number;
}

export type MinMap = {
	MapID: string;

	LastUpdate: string;

	Name: string;
};
